src/TreeHelpers/TBranchCollection.cpp

```diff
@@ -12,60 +12,66 @@ using namespace std;
 
 namespace BAT {
 
-void TBranchCollection::addBranchToTree(std::string branchName, std::string branchLabel, TreePointer tree) {
+void TBranchCollection::addBranchToTree(std::string branchLabel, std::string varType, TreePointer tree, bool isSingleValuePerEvent) {
```

Comment:
Note: I think if you template this function you should be able to add arbitrary types to the Tree if it is supported by TTree